hicommonwealth/commonwealth
# Only show gated topic banner for users with restricted memberships

## Link to Issue
Closes: 
- https://github.com/hicommonwealth/commonwealth/issues/5952
- https://github.com/hicommonwealth/commonwealth/issues/5951

## Description of Changes
- Now only showing gated topic banner in view thread page for users with restricted memberships
- Polls interaction is blocked on view threads page if the thread belongs to a restricted group topic

## "How We Fixed It"
N/A

## Test Plan

Test if the gating banner is only showed when user is in restricted group topic
- Be in a community in which you are also in a Group.
- Go to a Topic gated by a group you are in.
- Click on any thread to view Thread
- Verify that you don't see the gated topic banner.
- Now go to a topic which is also gated and you are not a part of that group
- Visit any thread of that topic
- Verify you see the gated topic banner.

Test if the polls interaction is disabled when user is in restricted group topic
- Be in a community in which you are not in a Group.
- Go to a Topic gated by a group you are not in.
- Find a thread that has polls and click on it
- View that you can't interact with the poll options and vote button and they are disabled. Hovering on the "Vote" button should show "Topic is gated"


## Deployment Plan
N/A

## Other Considerations
N/A
